internal/objects/job/job.go

```diff
@@ -42,7 +42,12 @@ const (
 func certgenJobName(contourImage string) string {
 	// [TODO] danehans: Remove and use contour.Name + "-certgen" when
 	// https://github.com/projectcontour/contour/issues/2122 is fixed.
-	return "contour-certgen-" + objutil.TagFromImage(contourImage)
+	res := "contour-certgen-" + objutil.TagFromImage(contourImage)
+	if len(res) > 63 {
+		// Truncate to avoid issues with long tags and K8s character limit
+		res = res[0:63]
```

Comment:
Another option I've seen used for this is to compute a hash of the full value and use that when needed, but I think this should work fine.